### PR TITLE
feat(emp-tools): add uniswap query and seperate perp components

### PR DESCRIPTION
**Pull request by daywiss**
# Motivation
#204 

# Summary
This is a continuation of perp support in EMP.  It adds uniswap graphql queries and some basic scaffolding for new perp components. Some components shared between EMP and Perp have been moved into features/core.

# Details
This PR seperates logic between Perp and EMP by moving perp specific components into features/perpetual. Much of this is copied from existing EMP components and will be modified in future PRs to get the functionality we need specific to perp.  There may be more code duplication than desired, but this poses little risk to breaking working EMP code and can be refactored later to refine component sharing. 

# Screenshots
No wallet connected
image

No contract selected
image

Perpetual selected. Note shared "totals" component, and "general info". Tabs have been modified to only show "general Info" until rest of components are built out.
image

### CODE REVIEW

File: features/perpetual/contract-state/GeneralInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,191 @@
+import styled from "styled-components";
```
Comment (daywiss) [inline features/perpetual/contract-state/GeneralInfo.tsx:1]: this file is directly copied from Emp contract-state/GeneralInfo. no review required really

File: features/perpetual/contract-state/ContractState.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { Box, Grid, Typography } from "@material-ui/core";
```
Comment (daywiss) [inline features/perpetual/contract-state/ContractState.tsx:1]: this file directly copied from emp/contractState. no review required

File: pages/index.tsx [hunk 3]
```diff
@@ -102,98 +210,134 @@ export default function Index() {
...
+        <Typography>Please connect your wallet before continuing.</Typography>
+      )}
+    </Blurb>
+  );
+}
+
+export default function Index() {
+  const { contract, isValid } = SelectedContract.useContainer();
+  // Use callback here will prevent re mounting components unless contract changes
+  const selectPage = useCallback(() => {
```
Comment (daywiss) [inline pages/index.tsx:322]: This index component now switches between 3 states. Perp contract selcted, Emp contract selected, and no/invalid contract selected.

File: containers/Uniswap.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { useState, useEffect } from "react";
+
+import { useQuery } from "@apollo/client";
+import { PAIR } from "../apollo/uniswap/queries";
+
+const useUniswap = ({ token0, token1 }: { token0: string; token1: string }) => {
```
Comment (mrice32) [inline containers/Uniswap.ts:6]: Is this used anywhere?

File: containers/SelectedContract.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  useEffect(() => {
+    // we need to wait until the contract list is loaded
+    if (loading) return;
+    const queryAddress = router.query.address;
+    const isNewAddress = queryAddress !== state.address;
+
+    if (queryAddress && isNewAddress && typeof queryAddress === "string") {
+      setAddress(queryAddress);
+    }
+  }, [router!.query!.address, loading]);
```
Comment (mrice32) [inline containers/SelectedContract.ts:26]: Interesting -- never seen this syntax before.

**Review by mrice32 — APPROVED** [review]
Looks good!

File: containers/Uniswap.ts (hunk 4 above)
Comment (daywiss) [reply containers/Uniswap.ts:6]: not used yet, but it will be to get our "fair value" for the perp

File: containers/SelectedContract.ts (hunk 5 above)
Comment (daywiss) [reply containers/SelectedContract.ts:26]: i think this is wrong actually. was going for[ optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)! dont know why this works though

**Pull request merged**